### PR TITLE
Redirect to the page user was previously on when they log in or sign up

### PR DESCRIPTION
**Pull request by kntsoriano**

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -41,4 +42,17 @@ def disable_site_header!
   def self.disable_site_header!(*args)
     before_action :disable_site_header!, *args
   end
+
+  private
+  def storable_location?
+    request.get? && is_navigational_format? && !devise_controller? && !request.xhr?
```
Comment (iHiD) [inline app/controllers/application_controller.rb:48]: We don't need to store this for logged in users. So you can add `&& !user_signed_in?`

File: app/controllers/user_onboarding_controller.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 class UserOnboardingController < ApplicationController
   skip_before_action :ensure_onboarded!
+  skip_before_action :store_user_location!
```
Comment (iHiD) [inline app/controllers/user_onboarding_controller.rb:3]: If we add the `user_signed_in?` bit as per my comment above, we won't need this here.

**Review by iHiD — APPROVED** [review]
Good work. Thanks dude. One small enhancement suggested.

**Comment by kntsoriano** [comment]
The session is cleared when user logs in so this won't work.

**Comment by iHiD** [comment]
OK, cool. I wonder if this is broken in v2 then: https://github.com/exercism/website/blob/master/app/controllers/application_controller.rb#L28

**Pull request merged**